### PR TITLE
OL-9

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: config/install/core.entity_form_display.node.challenge.default.yml [hunk 1]
```diff
@@ -6,6 +6,7 @@ dependencies:
     - field.field.node.challenge.body
     - field.field.node.challenge.field_category
     - field.field.node.challenge.field_close_on
+    - field.field.node.challenge.field_comments
```
Comment (db-Matroskeen) [inline config/install/core.entity_form_display.node.challenge.default.yml:9]: I'm not sure that we need comments in "Challenge". Can we clarify this with Zohar?

File: modules/openideal_challenge/openideal_challenge.module [hunk 2]
```diff
@@ -73,3 +75,80 @@ function openideal_challenge_views_query_alter(ViewExecutable $view, QueryPlugin
...
+    'tokens' => [
+      'openideal' => $count_of_ideas,
+    ],
+  ];
+}
+
+/**
+ * Implements hook_tokens().
+ */
+function openideal_challenge_tokens($type, $tokens, array $data, array $options, BubbleableMetadata $bubbleable_metadata) {
```
Comment (db-Matroskeen) [inline modules/openideal_challenge/openideal_challenge.module]: Can we move token hooks to a separate *.tokens.inc file?

File: modules/openideal_challenge/openideal_challenge.module [hunk 3]
```diff
@@ -73,3 +75,80 @@ function openideal_challenge_views_query_alter(ViewExecutable $view, QueryPlugin
...
+  if ($type == 'openideal' && !empty($data['challenge_id'])) {
+    $challenge_id = $data['challenge_id']->jsonSerialize();
+    $entity_type_manager = \Drupal::entityTypeManager()->getStorage('node');
+    /** @var \Drupal\node\Entity\Node $challenge */
+    $challenge = $entity_type_manager->load((int) $challenge_id);
+
+    foreach ($tokens as $name => $original) {
+      switch ($name) {
+        case 'count-of-ideas':
+          $count = _openideal_challenge_get_count_of_ideas($challenge->id());
```
Comment (db-Matroskeen) [inline modules/openideal_challenge/openideal_challenge.module]: Variable "$count" seems redundant.

File: modules/openideal_challenge/openideal_challenge.module [hunk 4]
```diff
@@ -73,3 +75,80 @@ function openideal_challenge_views_query_alter(ViewExecutable $view, QueryPlugin
...
+ * @throws \Drupal\Component\Plugin\Exception\PluginNotFoundException
+ */
+function _openideal_challenge_get_count_of_ideas($challenge_id) {
+  $query = \Drupal::entityTypeManager()
+    ->getStorage('node')
+    ->getQuery();
+  $query->condition('type', 'idea');
+  $query->condition('status', NodeInterface::PUBLISHED);
+  $query->condition('field_challenge', $challenge_id);
+  $query->accessCheck(FALSE);
```
Comment (db-Matroskeen) [inline modules/openideal_challenge/openideal_challenge.module]: I think it makes sense to keep access checks here. Users should be able to see the number of visible ideas.

File: modules/openideal_challenge/openideal_challenge.module [hunk 5]
```diff
@@ -73,3 +75,80 @@ function openideal_challenge_views_query_alter(ViewExecutable $view, QueryPlugin
...
+ * @param string $challenge_id
+ *   Challenge id.
+ *
+ * @return string
+ *   THe count of ideas.
+ *
+ * @throws \Drupal\Component\Plugin\Exception\InvalidPluginDefinitionException
+ * @throws \Drupal\Component\Plugin\Exception\PluginNotFoundException
+ */
+function _openideal_challenge_get_count_of_ideas($challenge_id) {
```
Comment (db-Matroskeen) [inline modules/openideal_challenge/openideal_challenge.module]: I would recommend creating a separate service with such methods (I think we'll have more in the future).

**Review by db-Matroskeen — CHANGES_REQUESTED** [review]

File: config/install/core.entity_form_display.node.challenge.default.yml (hunk 1 above)
Comment (ghost) [reply config/install/core.entity_form_display.node.challenge.default.yml:9]: That's right, Zohar confirmed that.

File: modules/openideal_challenge/openideal_challenge.module (hunk 2 above)
Comment (ghost) [reply modules/openideal_challenge/openideal_challenge.module]: Thanks, it's done.

File: modules/openideal_challenge/openideal_challenge.module (hunk 4 above)
Comment (ghost) [reply modules/openideal_challenge/openideal_challenge.module]: Done.

File: modules/openideal_challenge/openideal_challenge.module (hunk 5 above)
Comment (ghost) [reply modules/openideal_challenge/openideal_challenge.module]: It makes sense, thanks.
I moved it to the existing challenge service.

**Pull request merged**